hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/bucket/BucketCache.java

```diff
@@ -999,6 +1019,12 @@ void freeSpace(final String why) {
         }
       }
 
+      // Check if the cold file eviction is sufficient to create enough space.
```

Comment:
@wchevreuil, right now in the code, I am evicting all the cold files and then checking if we have more bytes to free.
Would it make sense to do it like this or do we want to evict only upto bytesToFreeWithExtra and exit leaving behind some of the cold blocks?
Which one would be preferable? The later option seems user friendly.

Reply:
I don't know, just wanted to make sure we considered the two options. If we delete all cold files even if that meant we will free more than the originally wanted free bytes, we might delay further mass evictions, at the cost of having more misses if these cold blocks are still accessed. I leave the final decision to you.

Reply:
Ok, given that we do not want to delay the current caching thread any further once the required space is available. This will also avoid misses on the remaining cold blocks. Hence, I think, return as soon as the the required space is available.